spring-integration-core/src/test/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiatorTests.java

```diff
@@ -284,17 +290,41 @@ public void testExceptionFromLock() throws Exception {
 
 		CountDownLatch onGranted = new CountDownLatch(1);
 
-		LockRegistryLeaderInitiator initiator = new LockRegistryLeaderInitiator(registry);
+		LockRegistryLeaderInitiator another = new LockRegistryLeaderInitiator(registry);
 
-		initiator.setLeaderEventPublisher(new CountingPublisher(onGranted));
+		another.setLeaderEventPublisher(new CountingPublisher(onGranted));
 
-		initiator.start();
+		another.start();
 
 		assertTrue(onGranted.await(10, TimeUnit.SECONDS));
-		assertTrue(initiator.getContext().isLeader());
+		assertTrue(another.getContext().isLeader());
 		assertTrue(exceptionThrown.get());
 
-		initiator.stop();
+		another.stop();
+	}
+
+	@Test
+	public void shouldShutdownInternalExecutorService() throws Exception {
+
+		this.initiator.start();
+		this.initiator.destroy();
```

Comment:
I think there is no guarantee in the method execution order in one test class.
So, this `destroy()` on the shared variable is going to be dangerous when this test method will be called before any other.

Does it make sense to you?

Reply:
JUnit instantiates an instance of the test class, per test method, and as none of the variables in the test are static there are no variables shared between tests

Reply:
OK. I see what you mean. Good to know 😄